diagnostics/diagnostic.go

```diff
@@ -68,6 +73,88 @@ func interfaceToJSONString(i interface{}) string {
 	return string(b)
 }*/
 
+func (d *DiagnosticClient) getSysInfo() {
+	ramInfo := GetRAMInfo()
+	diskInfo := GetDiskInfo()
+	cpuInfo := GetCPUInfo()
+
+	d.mu.Lock()
+	d.hardwareInfo = diaglib.HardwareInfo{
+		RAM:  ramInfo,
+		Disk: diskInfo,
+		CPU:  cpuInfo,
+	}
+	d.mu.Unlock()
+}
+
+func GetRAMInfo() diaglib.RAMInfo {
+	totalRAM := uint64(0)
+	freeRAM := uint64(0)
+
+	vmStat, err := mem.VirtualMemory()
+	if err == nil {
+		totalRAM = vmStat.Total
+		freeRAM = vmStat.Free
+	}
+
+	return diaglib.RAMInfo{
+		Total: totalRAM,
+		Free:  freeRAM,
+	}
+}
+
+func GetDiskInfo() diaglib.DiskInfo {
+	fsType := ""
+	total := uint64(0)
+	free := uint64(0)
+
+	partitions, err := disk.Partitions(false)
+
+	if err == nil {
+		for _, partition := range partitions {
+			if partition.Mountpoint == "/" {
+				iocounters, err := disk.Usage(partition.Mountpoint)
```

Comment:
need somehow understand: on which disk `datadir/chaindata`/`datadir/snapshots`/`datadir/snapshots/domain` folders? (for example if it symlinked to another disk)

Reply:
https://github.com/ledgerwatch/erigon/pull/9505